File: CSharp/ImageProcessing/ComOpenFileDialog.cs
using Microsoft.Win32;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ImageProcessing
{
    public class ComOpenFileDialog : ComFileDialog
    {
        public ComOpenFileDialog() : base()
        {
        }

        ~ComOpenFileDialog()
        {
        }

        public override bool ShowDialog()
        {
            bool bRst = false;

            OpenFileDialog openFileDlg = new OpenFileDialog();
            openFileDlg.FileName = base.FileName;
            openFileDlg.InitialDirectory = base.InitialDirectory;
            openFileDlg.Filter = base.Filter;
            openFileDlg.FilterIndex = base.FilterIndex;
            openFileDlg.Title = base.Title;
            openFileDlg.CheckFileExists = base.CheckFileExists;
            openFileDlg.CheckPathExists = base.CheckPathExists;
            if (openFileDlg.ShowDialog() == true)
            {
                base.m_strFilePass = openFileDlg.FileName;
                bRst = true;
            }

            return bRst;
        }
    }
}

File: Vb/ImageProcessing/ComOpenFileDialog.vb
Imports Microsoft.Win32

Namespace ImageProcessing
    Public Class ComOpenFileDialog : Inherits ComFileDialog
        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overloads Sub Finalize()
            MyBase.Finalize()
        End Sub

        Public Overrides Function ShowDialog() As Boolean
            Dim bRst As Boolean = False
            Dim openDialog As OpenFileDialog = New OpenFileDialog()
            openDialog.FileName = MyBase.FileName
            openDialog.InitialDirectory = MyBase.InitialDirectory
            openDialog.Filter = MyBase.Filter
            openDialog.FilterIndex = MyBase.FilterIndex
            openDialog.Title = MyBase.Title
            openDialog.CheckFileExists = MyBase.CheckFileExists
            openDialog.CheckPathExists = MyBase.CheckPathExists
            If (openDialog.ShowDialog() = True) Then
                MyBase.m_strFilePass = openDialog.FileName
                bRst = True
            End If

            Return bRst
        End Function
    End Class
End Namespace